File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double maxSum = 0;
        double curr = 0;
        for (int i = 0; i < k; i++) {
            curr += nums[i];
        }
        maxSum = curr; 
        int left = 0;
        for (int right = k; right < nums.length; right++) {
            curr += nums[right] - nums[right - k];
            maxSum = Math.max(curr, maxSum);
        }
        return maxSum / (double) k;
    }
}

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
       
        curr = sum(nums[:k])
        answer = curr
        for i in range(k, len(nums)):
            curr += nums[i ] - nums[i - k]

            answer = max(answer, curr)
        
        return float(answer) / k 